theme/static/js/hs_file_browser.js

```diff
@@ -902,6 +933,8 @@ function bindFileBrowserItemEvents() {
                         }
 
                         if (aggrAppKey && toolAppKey) {
+                            // both the tool and aggregation have appkey restriction
+                            // both app keys must match for the tool to be available
                             if (aggrAppKey === toolAppKey) {
                                 appKeyApp = true;
```

Comment:
does this L939 `appKeyApp = true;` ever get executed?
Because `aggrAppKey === toolAppKey` will only be truthy when `extensionApp` is truthy (see L897 above where we check these keys). But this if block only gets executed when `!extensionApp` (see line 922). 

Oh man, these nested if/else blocks are making my head hurt! :)

Reply:
L939 will be executed. `aggrAppKey ` is not dependent on `extensionApp`.  And we compute  `toolAppKey` both when `extensionApp` is True or False.